Topics/Software_Tools/Git/Git_Merging_and_Rebasing.md

```diff
@@ -11,4 +11,118 @@ Both `git merge` and `git rebase` are methods for integrating new commits that a
\ No newline at end of file
+    If you prefer to read an article over watching a video, this is the resource for you, as it covers all of the topics that the above two videos cover. It also covers more scenarios that you may find yourself in while using rebase as part of your workflow. In particular, the section called [The Golden Rule of Rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing) will help you make sure that you don't accidentally bring catastrophe upon your repository while using rebase. Additionally, there is a short section on interactive rebase, a powerful tool that give you control over your branch history. 
+
+
+# A primer on `git merge` and `git rebase`
+
+One of the most useful features that Git introduced was *branching*. This is the cornerstone of how collaboration is done in Git, but one of the complications is when you want to reconcile the changes made in one branch with the work done in another branch. 
+
+You may be familiar with the most common use case: you want to update your branch with the latest changes from `main`.
+
+There are two different ways of combining two branches: merging and rebasing. They work quite differently, and we’ll explore when and why you’d choose one over the other.
+
+## Merging
+
+The first and classic way to combine the work done in two branches is to *merge* the two branches together. Merging is usually the default way to reconcile changes between branches because it preserves commit history (and is thus “safer”), and resolving merge conflicts is (relatively) simple compared to rebasing.
+
+Say you are currently checked out into the `feature` branch, and you want to merge your commits in that branch into the `main` branch.
+
+Your `feature` branch branched off of the `main` branch at some commit. We’ll illustrate the commits in chronological order (so commit B was committed after commit A).
+
+We’ll say that feature branched off of `main` at some commit C.
+
+So, your `main` branch may look like:
+> A -> B -> C -> E -> F
+
+And, your `feature` branch may look like:
+> A -> B -> C -> D -> G
+
+But notice that there are commits in main that happened after feature was branched off. These are commits E and F. We want to merge the changes made in those commits into `feature`; perhaps they are useful for the work we’re doing in `feature`. 
+
+What we’ll do is merge `main` into `feature`. This firstly creates a merge commit in `feature` which incorporates all of the changes made in `main` but not in `feature` by comparing their *diffs*. Any changes made in `main` but not `feature` are then applied to `feature` via the merge commit.
```

Comment:
Might be helpful to explicitly state the command being used: `git merge main` while we're checked into the `feature` branch.

Reply:
Good idea, it could be confusing. I've added a code snippet for this.